astroid/interpreter/_import/spec.py

```diff
@@ -201,7 +201,12 @@ def find_module(
         if processed:
             modname = ".".join(processed + [modname])
         if util.is_namespace(modname) and modname in sys.modules:
-            submodule_path = sys.modules[modname].__path__
+            try:
```

Comment:
Do we understand why `__path__` isn't available? 

https://docs.python.org/3/reference/import.html#path__
`If the module is a package (either regular or namespace), the module object’s __path__ attribute must be set.`

We probably shouldn't crash, but is this "normal" behaviour?

https://docs.python.org/3/library/importlib.html#importlib.machinery.ModuleSpec
`Thus it is possible to update the module’s __path__ at runtime, and this will not be automatically reflected in __spec__.submodule_search_locations.`

`__path__` not being available seems to indicate that this isn't a package and thus perhaps we shouldn't return a `ModuleSpec` with a `PY_NAMESPACE` type?

Reply:
Just dashing off a quick reply that when I looked into it I thought it was because `urllib3` was doing magic with `six`, so I wasn't concerned, but I can try to get a real answer :-)

Reply:
Coming back to this, I guess I don't want to take on the burden of understanding `_SixMetaPathImporter` inside and out -- there are four bugs filed against it anyway, so I wouldn't trust it to work in all scenarios, esp. if libraries like `urllib3` are doing things on top of it. I'd rather just catch the error until someone can demonstrate that `astroid` is at fault.

Reply:
Sometime the code is too dynamic to be reasonably understood by astroid. Understanding this kind of dynamic code would be very good and reduce the use of brain and workarounds but I agree with @jacobtylerwalls. Ready to release the next patch version of astroid as soon as we merge this :)

Reply:
Note that this is still on my radar but I haven't been around much. Should be able to get to this in a 10 days or so. My main question is whether the check for `__path__` should be included in the `is_namespace` function.

Reply:
Thanks Daniel, looking again, I think you have a point. I pushed a new commit to fix this in `is_namespace()`. ModuleSpec has `submodule_search_locations` as a better name than `__path__` per [PEP 451](https://peps.python.org/pep-0451/#attributes).